### DISCUSSION TITLE
Slow performance of model when exported in other formats like .tflite

### DISCUSSION THREAD

**Discussion by UsamaRehman001** [ultralytics/ultralytics#10222] [Q&A]
I would like to ask about how can i improve the model performance?. When i use .pt file it process approximately 10 frames per seconds, on average each frame with detection or no detection take almost 80ms to 90ms. But when i export into tflite it take even less frames per seconds and on average process each frame with 250ms with detection or no detection. What could be wrong?  Also the same .pt file dosen't work well on rasberry pi 5. it process like 3 frames when no detection is in frame, and fps even drop to 1 or frames when a detection is found. How to resolve this issue.

**Comment by pderrenger** [comment]
Hey there! 😊 It sounds like you're running into performance issues after exporting your YOLOv8 model to `.tflite` and struggling with performance on Raspberry Pi 5. Let's see how we can address this.

For `.tflite`, the performance can vary significantly depending on the device hardware and the optimizations applied during the conversion process. Here are a couple of quick suggestions:

1. **Quantization**: If not already done, consider using quantization during export, which can reduce model size and potentially increase performance at the cost of some accuracy. For `tflite`, you can use the `int8` quantization, which is especially beneficial for edge devices.
   ```bash
   yolo export model=yolov8n.pt format=tflite int8=True
   ```

2. **Optimized Operations**: Ensure that your TFLite runtime is using optimized operations where available. TensorFlow Lite supports using optimized kernels with the XNNPACK delegate, which can offer speed improvements.

For Raspberry Pi 5, the performance can be bottlenecked by the CPU capabilities, especially for high-complexity models:

- **Model Simplification**: Consider using a smaller model variant or simplifying your model if possible.
- **Edge Optimizations**: Utilizing optimizations specific to the ARM architecture on Raspberry Pi, such as using the Coral Edge TPU with a compatible model format, can provide significant speedups.

It's also important to remember that performance can be highly dependent on the specific setup and use case. Trial and adjustment might be required to find the best configuration for your needs.

Feel free to share more details or configurations you've tried for potentially more targeted advice. Good luck with your optimization efforts! 🚀
